pkg/apis/extensions/v1beta1/types.go

```diff
@@ -1102,3 +1103,78 @@ type PodSecurityPolicyList struct {
 	// Items is a list of schema objects.
 	Items []PodSecurityPolicy `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// +genclient=true
+
+// Template contains the inputs needed to produce a Config.
+type Template struct {
+	unversioned.TypeMeta `json:",inline"`
+	v1.ObjectMeta        `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec TemplateSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+}
+
+type TemplateSpec struct {
+	// Optional: Parameters is an array of Parameters used during the
+	// Template to Config transformation.
+	Parameters []Parameter `json:"parameters,omitempty" protobuf:"bytes,2,rep,name=parameters"`
+
+	// Required: A yaml config list of resources to create
```

Comment:
Thinking on this further, this may not be required to say at all.  The Objects are just Objects.  It is a bit weird since here they are RawExtensions and have a '[]bytes raw' field, but I think it is clear enough.